lib/kafka-producer.js

```diff
@@ -70,23 +69,30 @@ Producer.prototype.getProducer = Promise.method(function (opts, topts) {
  * @private
  */
 Producer.prototype._serializeType = function (topicName, isKey, data) {
-  var schema = isKey === false
-    ? this.sr.valueSchemas[topicName]
-    : this.sr.keySchemas[topicName];
+  // implementing io.confluent.kafka.serializers.subject.TopicRecordNameStrategy
+  // FIXME: It will break if code gets minimized
```

Comment:
why the code would be minimized?

Reply:
What I meant is if you use this library in your project and minimize the project and its dependencies. It'll break because the the type name will be converted to a random string.